### PR TITLE
Upgrade Kind version & Add support for multiple CRI in kinder

### PR DESCRIPTION
**Pull request by fabriziopandini**
This PR:
- upgrade kind library to the latest version (v0.4.0)
- introduces in kind support for multiple CRI, currently with docker and containerd implemented, thus allowing to preserve a test signal with docker
- changes kinder e2e workflows to continue to use docker
- aligns action implementation in kinder to kind (no more task, join using config files instead of flags)
- simplifies and improve consistent actions output

All the e2e workflows are passing locally after those changes.

Ps. I apologize for the size of the PR. 
In case, I'm open to suggestion to break down the change in multiple PR.
ATM, in order to reduce a little bit the pain of reviewers, I grouped files in commits

### CODE REVIEW

File: kinder/pkg/cluster/status/node.go [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+*/
+
+package status
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
```
Comment (neolit123) [inline kinder/pkg/cluster/status/node.go]: `sigs.k8s.io/yaml`

File: kinder/cmd/kinder/get/get.go [hunk 2]
```diff
@@ -14,35 +14,33 @@ See the License for the specific language governing permissions and
...
 
 import (
 	"github.com/spf13/cobra"
 
-	"sigs.k8s.io/kind/cmd/kind/get/clusters"
-	"sigs.k8s.io/kind/cmd/kind/get/kubeconfigpath"
-	"sigs.k8s.io/kind/cmd/kind/get/nodes"
+	kindCluster "sigs.k8s.io/kind/cmd/kind/get/clusters"
+	kindKubeConfig "sigs.k8s.io/kind/cmd/kind/get/kubeconfigpath"
+	kindNodes "sigs.k8s.io/kind/cmd/kind/get/nodes"
```
Comment (neolit123) [inline kinder/cmd/kinder/get/get.go]: we should change to e.g. kindnodes

File: kinder/cmd/kinder/build/nodevariant/nodevariant.go [hunk 3]
```diff
@@ -92,10 +99,18 @@ func NewCommand() *cobra.Command {
 }
 
 func runE(flags *flagpole, cmd *cobra.Command, args []string) error {
+	// check the cri flag
+	var err error
+	cri := flags.Cri
```
Comment (neolit123) [inline kinder/cmd/kinder/build/nodevariant/nodevariant.go]: possibly `CRI`?

File: kinder/ci/workflows/external-etcd.yaml [hunk 4]
```diff
@@ -52,7 +55,6 @@ tasks:
     - kubeadm-init
     - --automatic-copy-certs
     - --name={{ .vars.clusterName }}
-    - --loglevel=debug
```
Comment (neolit123) [inline kinder/ci/workflows/external-etcd.yaml]: better to have loglevel=debug,
also in the other workflows too?

File: kinder/pkg/cluster/status/node.go (hunk 1 above)
Comment (fabriziopandini) [reply kinder/pkg/cluster/status/node.go]: done

File: kinder/cmd/kinder/get/get.go (hunk 2 above)
Comment (fabriziopandini) [reply kinder/cmd/kinder/get/get.go]: done

File: kinder/cmd/kinder/build/nodevariant/nodevariant.go (hunk 3 above)
Comment (fabriziopandini) [reply kinder/cmd/kinder/build/nodevariant/nodevariant.go]: done

**Comment by fabriziopandini** [comment]
@neolit123 thanks for the feedback!
everything should be addressed now

**Comment by neolit123** [comment]
/lgtm
thanks.

**Pull request merged**